src/common/rng/xoroshiro128.c

```diff
@@ -0,0 +1,70 @@
+
+/*****************************************************************************\
+ *                                                                           *\
+ *  This file is part of the Verificarlo project,                            *\
+ *  under the Apache License v2.0 with LLVM Exceptions.                      *\
+ *  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception.                 *\
+ *  See https://llvm.org/LICENSE.txt for license information.                *\
+ *                                                                           *\
+ *                                                                           *\
+ *  Copyright (c) 2022                                                       *\
+ *     Verificarlo Contributors                                              *\
+ *                                                                           *\
+ ****************************************************************************/
+
+#include <stdint.h>
+
+#include "splitmix64.h"
+#include "xoroshiro128.h"
+
+/*********************** XOROSHIRO128++ FUNCTIONS ****************************\
+ *                                                                           *\
+ *  Written in 2019 by David Blackman and Sebastiano Vigna (vigna@acm.org)   *\
+ *                                                                           *\
+ *  To the extent possible under law, the author has dedicated all copyright *\
+ *  and related and neighboring rights to this software to the public domain *\
+ *  worldwide. This software is distributed without any warranty.            *\
+ *                                                                           *\
+ *  See <http://creativecommons.org/publicdomain/zero/1.0/>.                 *\
```

Comment:
is this license compatible with apache2.0? I suppose yes, but we need to be 100% positive.

Reply:
I checked and normally it is the most [permissive license](https://creativecommons.org/publicdomain/zero/1.0/)

> The person who associated a work with this deed has dedicated the work to the public domain by waiving all of his or her rights to the work worldwide under copyright law, including all related and neighboring rights, to the extent allowed by law.
> 
> You can copy, modify, distribute and perform the work, even for commercial purposes, all without asking permission. See Other Information below.

Reply:
It seems like CC in Apache2.0 program is doable but not the opposite. It seems to be more complicated than expected matter. I will try to find a qualified person to give a definitive answer, especially if we can sublicence, if we need to retain header, license file for that part and so on..
You enclosed the CC paragraph in a file starting with our apache2.0 header, and did not put a marker when the CC code end. Maybe the correct solution is to remove the apache header, move up the CC header at the begining of the file and have this file being CC... 
@pablooliveira any opinion/advice on your side?

Reply:
This is CC0  so equivalent to a Public domain license. Should be fine to include it. You can keep the original header.

Reply:
@pablooliveira to be sure we are on the same page, you also suggest to remove our apache header from this file and keep only the CC header?

Reply:
@titeup yes, that seems good for me. Keep only the original CC0 header on that file.